File: Easy/Winner of an election/winner-of-an-election.py
class Solution:
    def winner(self, arr, n):
        d = {}
        for i in arr:
            if i not in d:
                d[i] = 1
            else:
                d[i] += 1

        max_votes = max(d.values())
        winners = [k for k, v in d.items() if v == max_votes]
        
        # Sort the winners lexicographically and pick the first one
        winner_name = min(winners)
        
        return [winner_name, str(max_votes)]


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__=="__main__":
    T=int(input())
    for _ in range(T):
        
        n=int(input())
        arr=input().strip().split()
        
        result = Solution().winner(arr,n)
        print(result[0],result[1])
# } Driver Code Ends

File: Easy/Winner of an election/winner-of-an-election.cpp
//{ Driver Code Starts
#include <iostream>
#include <map>
#include <algorithm>
#include <cstdlib>
#include<bits/stdc++.h>

using namespace std;


// } Driver Code Ends

#include <bits/stdc++.h>
using namespace std;
class Solution{
public:
    //Function to return the name of candidate that received maximum votes.
    vector<string> winner(string arr[],int n){
      unordered_map<string, int> map;
      for(int i=0;i<n;i++){
        map[arr[i]]++;
      }
      int maxvotes = -1;
      string res;
      for (auto x : map){
        if (x.second > maxvotes){
          maxvotes = x.second;
          res = x.first;
        }else if(x.second==maxvotes){
          res = min(res, x.first);
        }
      }
      return {res,to_string(maxvotes)};
    }
};

//{ Driver Code Starts.

int main()
{
    int t;
    cin>>t;
    
    for(int i=0;i<t;i++)
    {
        
        
        int n;
        cin>>n;
        
        string arr[n];
        
        for (int i=0;i<n;i++)
        cin>>arr[i];
        Solution obj;
        vector<string> result = obj.winner(arr,n);
        
        cout<<result[0] << " " << result[1] << endl;
    }
    return 0;
}

// } Driver Code Ends